qualtran/bloqs/rotations/phase_gradient.ipynb

```diff
@@ -0,0 +1,531 @@
+   "metadata": {
+    "cq.autogen": "AddScaledValIntoPhaseReg.bloq_doc.py"
+   },
+   "outputs": [],
+   "source": [
+    "from qualtran.bloqs.rotations.phase_gradient import AddIntoPhaseGrad"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ff3bc5ed",
+   "metadata": {
+    "cq.autogen": "AddScaledValIntoPhaseReg.example_instances.md"
+   },
+   "source": [
+    "### Example Instances"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "18a47d9c",
+   "metadata": {
+    "cq.autogen": "AddScaledValIntoPhaseReg.add_scaled_val_into_phase_reg_example"
+   },
+   "outputs": [],
+   "source": [
+    "add_unitary = AddScaledValIntoPhaseReg(QFxp(2, 2), phase_bitsize=2, gamma=2, gamma_dtype=QFxp(2, 2))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "10b1c40e",
+   "metadata": {
+    "cq.autogen": "AddScaledValIntoPhaseReg.graphical_signature.md"
+   },
+   "source": [
+    "#### Graphical Signature"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "584ab205",
+   "metadata": {
+    "cq.autogen": "AddScaledValIntoPhaseReg.graphical_signature.py"
+   },
+   "outputs": [],
+   "source": [
+    "from qualtran.drawing import show_bloqs\n",
+    "show_bloqs([add_scaled_val_into_phase_reg],\n",
+    "           ['`add_scaled_val_into_phase_reg`'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "105f37bf",
+   "metadata": {
+    "cq.autogen": "AddScaledValIntoPhaseReg.call_graph.md"
+   },
+   "source": [
+    "### Call Graph"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "57ad18ba",
+   "metadata": {
+    "cq.autogen": "AddScaledValIntoPhaseReg.call_graph.py"
+   },
+   "outputs": [],
+   "source": [
+    "from qualtran.resource_counting.generalizers import ignore_split_join\n",
+    "add_scaled_val_into_phase_reg_g, add_scaled_val_into_phase_reg_sigma = add_scaled_val_into_phase_reg.call_graph(max_depth=1, generalizer=ignore_split_join)\n",
+    "show_call_graph(add_scaled_val_into_phase_reg_g)\n",
+    "show_counts_sigma(add_scaled_val_into_phase_reg_sigma)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d82675a2",
+   "metadata": {
+    "cq.autogen": "PhaseGradientState.phase_gradient_state"
+   },
+   "outputs": [],
+   "source": [
+    "phase_gradient_state = PhaseGradientState(4)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0984e1dd",
+   "metadata": {
+    "cq.autogen": "AddIntoPhaseGrad.add_into_phase_grad"
+   },
+   "outputs": [],
+   "source": [
+    "add_into_phase_grad = AddIntoPhaseGrad(4, 4)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d470659a",
+   "metadata": {
+    "cq.autogen": "AddScaledValIntoPhaseReg.add_scaled_val_into_phase_reg"
+   },
+   "outputs": [],
+   "source": [
+    "add_scaled_val_into_phase_reg = AddScaledValIntoPhaseReg(QFxp(2, 2), phase_bitsize=2, gamma=2, gamma_dtype=QFxp(2, 2))"
+   ]
+  }
```

Comment:
short version: these cells need to be manually moved up to their proper location within the file

long version: the `autogenerate-bloqs-notebooks-v2.py` script will automatically make cells in a notebook derived from the bloq examples, but sometimes we want to add additional cells with custom content. So if you re-run the autogenerate script it will only add/update the cells it's in charge of and will leave all the cells in place. It uses a piece of metadata (that you can see if you look at the json text within the `.ipynb` file) to tag each autogenerated cell with a key. If you change the name of the bloq example, it changes the key we use to look for the autogenerated cells and will add new cells to the end of the notebook and leave the vestigial ones in place. We could probably make it automatically delete the vestigial cells but it would probably be difficult add the new cells anywhere other than the bottom. But you can order them manually from within the jupyter notebook interface